src/app/workbench/useCommentPanel.ts

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { CommentPanel, defaultUploadComment, IMatchingCommentTemplate } from 'tdp_comments';
+
+export function useCommentPanel(selection: string[], itemIDType: string, commentsOpen: boolean): [(element: HTMLElement | null) => void, CommentPanel | null] {
+  const [instance, setInstance] = React.useState<CommentPanel | null>(null);
+
+  const setRef = React.useCallback(async (ref: HTMLElement | null) => {
+    setInstance((currentInstance) => {
+      // If the element ref did not change, do nothing.
+      if (currentInstance && ref) {
+        return currentInstance;
+      }
+
+      // Create a new one if there is a ref
+      if (ref) {
+        const panel = new CommentPanel({
+          appKey: 'reprovisyn',
+          commentTemplate: defaultUploadComment({ app_key: 'reprovisyn' }),
```

Comment:
Extract the app key to a constant since there are multiple occurrences in this file.

Reply:
Done!